cmd/thanos/compact.go

```diff
@@ -457,6 +460,19 @@ func runCompact(
 
 		return cleanPartialMarked()
 	}
+	g.Add(func() error {
+		ps := compact.NewDefaultPlanSim(grouper, planner, sy)
+		numberOfIterations, numberOfBlocksMerged, err := ps.Simulate(context.Background())
```

Comment:
Let's expose these results as Prometheus Gauge metrics

Reply:
A better way here is to only return an error in the `Simulate` method. 
We can add metrics definition to the PlanSim struct and update metrics value inside `Simulate` method.

Reply:
Thanks!
Just pushed a commit exporting the metrics. Let me know what you think!